lorien/UI/Dialogs/SettingsDialog.gd

```diff
@@ -161,3 +185,22 @@ func _on_OptionButton_item_selected(idx: int):
 	Settings.set_value(Settings.GENERAL_LANGUAGE, locale)
 	TranslationServer.set_locale(locale)
 	_general_restart_label.show()
+
+# -------------------------------------------------------------------------------------------------
+func _on_UIScaleOptions_item_selected(index: int):
+	match index:
+		UI_SCALE_AUTO_INDEX:
+			_ui_scale.set_editable(false)
+			Settings.set_value(Settings.APPEARANCE_UI_SCALE_MODE, Types.UIScale.AUTO)
+		UI_SCALE_CUSTOM_INDEX:
+			_ui_scale.set_editable(true)
+			Settings.set_value(Settings.APPEARANCE_UI_SCALE_MODE, Types.UIScale.CUSTOM)
+	emit_signal("ui_scale_changed")
+	popup_centered()
+
+# -------------------------------------------------------------------------------------------------
+func _on_UIScale_value_changed(value: float):
+	if Input.is_action_just_pressed("ui_accept") || _ui_scale._ready:
```

Comment:
`_ui_scale._ready` should be replaced with a public equivilant in `DelayedSpinBox.gd` (like a `is_ready()` method)